R/format-report.R

```diff
@@ -746,19 +783,29 @@ trace_matrix_notes <- function(exports_df){
 #' @param return_vals Logical (T/F). If `TRUE`, return the objects instead of printing them out for `rmarkdown`. Used for testing.
 #'
 #' @keywords internal
-format_appendix <- function(extra_notes_data, return_vals = FALSE){
-  sub_header_strs <- c("\n## R CMD Check\n\n", "\n## Test coverage\n\n")
+format_appendix <- function(extra_notes_data, return_vals = FALSE, scorecard_type = "R") {
+  check_title <- if (identical(scorecard_type, "R")) {
+    "R CMD Check"
+  } else {
+    "Check output"
+  }
+  sub_header_strs <- c(paste0("\n## ", check_title, "\n\n"), "\n## Test coverage\n\n")
 
   ### Covr Results ###
   # Format Table
-  covr_results_df <- extra_notes_data$covr_results_df
-  if (is.numeric(covr_results_df$test_coverage)) {
-    covr_results_df <- covr_results_df %>%
-      dplyr::mutate(test_coverage = paste0(.data$test_coverage, "%")) %>%
+  cov_results_df <- extra_notes_data$cov_results_df
```

Comment:
I looked into what happens if you pass an overall coverage score, but _not_ file-level scores. I used the `local_create_external_results()` helper and then modified the `.coverage.json` to only contain `{"overall":91.5426}`.

This still worked, and coverage displays correctly in the tables in the first few pages of the scorecard. Well done!

However, it still prints a coverage table at the bottom, that is empty:

![Screenshot 2024-08-09 at 5 30 06 PM](https://github.com/user-attachments/assets/bd103589-9efc-420b-a11a-5f2cc573e61f)

Could we check if there are any rows in this table before we render it? And then, if there aren't any, maybe replace the "Test coverage is calculated per script..." note with something like "Per file test coverage not provided." Something to make it clear that this is expected (based on the input) and isn't a rendering error.

Note: This can optionally be punted for a future release, because it isn't really relevant for R or Go packages, but I think it may be relevant down the line, as we get into other languages.

Reply:
> Could we check if there are any rows in this table before we render it? And then [...]

Sure, I'll make that change.

> Note: This can optionally be punted for a future release, because it isn't really relevant for R or Go packages, but I think it may be relevant down the line, as we get into other languages.

Hmm, the commit related to making coverage optional was added for an upcoming case where we know there will be no coverage, but I'm not sure how likely it is that a package could provide overall coverage but not per file.  Anyway, that's not an argument against the nicer handling you propose.